File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.py
count = 0
def solution(word):
    global count
    vowels = ['A', 'E', 'I', 'O', 'U']
    
    def dfs(word_string):
        global count
        count += 1
        
        if word_string == word: return True #찾음
        elif len(word_string) == 5: return False
        
        for v in vowels:
            if dfs(word_string + v): return True # if문 반환값이 True면 바로 count 반환

    for v in vowels:
        if dfs(v): return count

File: 프로그래머스/2/84512. 모음 사전/모음 사전.cpp
#include <string>
#include <vector>

using namespace std;

int count = 0;
bool found = false;
vector<char> vowels = {'A', 'E', 'I', 'O', 'U'};

bool dfs(string current, const string& target) {
    count++;

    if (current == target) {
        found = true;
        return true;
    }
    if (current.length() == 5) return false;

    for (char v : vowels) {
        if (dfs(current + v, target)) return true;
    }

    return false;
}

int solution(string word) {
    count = 0;
    found = false;

    for (char v : vowels) {
        if (dfs(string(1, v), word)) break;
    }

    return count;
}
